diff --git a/src/network/ssl/ssl.pri b/src/network/ssl/ssl.pri
index 230c45c26..d7527949e 100644
--- a/src/network/ssl/ssl.pri
+++ b/src/network/ssl/ssl.pri
@@ -117,7 +117,7 @@ qtConfig(ssl) {
 
         qtConfig(openssl-linked): {
             android {
-                build_pass|single_android_abi: LIBS_PRIVATE += -lssl_$${QT_ARCH} -lcrypto_$${QT_ARCH}
+                build_pass|single_android_abi: LIBS_PRIVATE += -lssl -lcrypto
             } else: QMAKE_USE_FOR_PRIVATE += openssl
         } else: \
             QMAKE_USE_FOR_PRIVATE += openssl/nolink
